#! /bin/sh -e

# DP: Allow the info docs to be built with emacs23.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        ;;
    *)
	echo >&2 "usage: `basename $0`: -patch|-unpatch [-d <srcdir>]"
        exit 1
esac
exit 0

# append the patch here and adjust the -p? flag in the patch calls.
--- Doc/info/Makefile~	2003-07-16 06:02:58.000000000 +0200
+++ Doc/info/Makefile	2007-08-15 12:06:41.000000000 +0200
@@ -34,10 +34,10 @@
 
 check-emacs-version:
 	@v="`$(EMACS) --version 2>&1 | egrep '^(GNU |X)Emacs [12]*'`"; \
-	if `echo "$$v" | grep '^GNU Emacs 2[12]' >/dev/null 2>&1`; then \
+	if `echo "$$v" | grep '^GNU Emacs 2[123]' >/dev/null 2>&1`; then \
 	  echo "Using $(EMACS) to build the info docs"; \
 	else \
-	  echo "GNU Emacs 21 or 22 is required to build the info docs"; \
+	  echo "GNU Emacs 21, 22, or 23 is required to build the info docs"; \
 	  echo "Found $$v"; \
 	  false; \
 	fi
